src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1700,6 +1699,19 @@ public void changed(Database.ChangeEvent event) {
             // Stop timeout timer:
             stopTimeout();
 
+            // In race condition, new doc or update doc is fired before starting to observe the
+            // DatabaseChangeEvent, it allows to skip few document changes with /_changes REST API.
+            // Make sure all document changes are tread by /_changes REST API.
+            if (!filled) {
+                filled = true;
+                RevisionList changes = db.changesSince(changesSince, changesOptions,
+                        changesFilter, changesFilterParams);
+                if (changes.size() > 0) {
+                    sendLongpollChanges(changes, changesSince);
```

Comment:
> Should it check if it's longpoll before calling sendLogpollChanges?

I believe it does not require to check it. DatabaseChangeListener is used for only feed=longpoll or continuous.

> Does it issue not apply to continuous feed?

I haven't seen that any CBL client replicator sends feed=continuous.